traitsui/tests/editors/test_html_editor.py

```diff
@@ -30,26 +32,205 @@ class HTMLModel(HasTraits):
     model_base_url = Str()
 
 
-def get_view(base_url_name, format_text=True, open_externally=False):
+def get_view(base_url_name):
     return View(
         Item(
             "content",
             editor=HTMLEditor(
-                format_text=format_text,
+                format_text=True,
                 base_url_name=base_url_name,
-                open_externally=open_externally
             )
         )
     )
 
 
+class HTMLContent:
+    """ Action to retrieve the HTML content currently displayed.
+    Implementation should return a str, whose content conforms to HTML markup.
+    """
+    pass
+
+
+def _is_webkit_page(page):
+    """ Return true if the given page is a QWebPage from QtWebKit.
+
+    Intended for handling the compatibility between QtWebKit and QtWebEngine.
+
+    Parameters
+    ----------
+    page : QWebEnginePage or QWebPage
+    """
+    return hasattr(page, "setLinkDelegationPolicy")
+
+
+def qt_get_page_html_content(page):
+    """ Return the HTML content currently being viewed.
+
+    Parameters
+    ----------
+    page : QWebEnginePage or QWebPage
+
+    Returns
+    -------
+    html : str
+    """
+    qt_allow_view_to_load(page)
+    if _is_webkit_page(page):
+        return page.mainFrame().toHtml()
+
+    content = []
+    page.toHtml(content.append)
+    qt_allow_view_to_load(page)
```

Comment:
This is because `QWebEnginePage.toHtml` is asynchronous. https://doc.qt.io/qt-5/qwebenginepage.html#toHtml
If this `qt_allow_view_to_load` is removed here, then the content returned will be empty.

Reply:
I think I see now, thanks